contracts/contracts/strategies/ConvexEthMetaStrategy.sol

```diff
@@ -263,16 +284,134 @@ contract ConvexEthMetaStrategy is InitializableAbstractStrategy {
         uint256 oethBalance = oeth.balanceOf(address(this));
         IVault(vaultAddress).burnForStrategy(oethBalance);
 
-        // Send all ETH and WETH on the contract, including extra
-        weth.deposit{ value: address(this).balance }();
+        // Get the strategy contract's ether balance.
+        // This includes all that was removed from the Metapool and
+        // any ether that was sitting in the strategy contract before the removal.
+        uint256 ethBalance = address(this).balance;
+        // Convert all the strategy contract's ether to WETH and transfer to the vault.
+        weth.deposit{ value: ethBalance }();
         require(
-            weth.transfer(vaultAddress, weth.balanceOf(address(this))),
+            weth.transfer(vaultAddress, ethBalance),
```

Comment:
This code changes no longer picks up any WETH if any was left on the contract. However that's okay since "depositAll" would put it into use.

Reply:
It is correct this change no longer picks up any WETH in the strategy contract for `withdraw` and `withdrawAll`. In normal operation there shouldn't be any WETH in the AMO strategy contract. We can see in practice that the AMO strategy contract has always had a zero balance of WETH. https://dune.com/queries/2738518/4557122?account_address_t6c1ea=0x1827F9eA98E0bf96550b2FC20F7233277FcD7E63

If for some reason there was WETH in the AMO strategy contract, for example someone transferred it some WETH, the `depositAll` will pick it up as DVF has pointed out.